core/capabilities/ccip/oraclecreator/bootstrap.go

```diff
@@ -104,5 +133,241 @@ func (i *bootstrapOracleCreator) Create(_ uint32, config cctypes.OCR3ConfigWithM
+	rmnHomeBoundContract := types.BoundContract{
+		Address: "0x" + hex.EncodeToString(config.Config.RmnHomeAddress),
+		Name:    consts.ContractNameRMNHome,
+	}
+
+	if err1 := i.contractReader.Bind(ctx, []types.BoundContract{rmnHomeBoundContract}); err1 != nil {
+		return nil, fmt.Errorf("failed to bind RMNHome contract: %w", err1)
+	}
+	rmnHomeReader := ccipreaderpkg.NewRMNHomePoller(
+		i.contractReader,
+		rmnHomeBoundContract,
+		i.lggr,
+		5*time.Second,
+	)
+	return rmnHomeReader, nil
+}
+
+// peerGroupDialer keeps watching for config changes and calls NewPeerGroup when needed.
+// Required for managing RMN related peer group connections.
+type peerGroupDialer struct {
+	lggr logger.Logger
+
+	peerGroupFactory rmn.PeerGroupFactory
+	rmnHomeReader    ccipreaderpkg.RMNHome
+
+	// common oracle config
+	bootstrapLocators  []commontypes.BootstrapperLocator
+	oraclePeerIDs      []ragep2ptypes.PeerID
+	commitConfigDigest [32]byte
+
+	activePeerGroups    []rmn.PeerGroup
+	activeConfigDigests []cciptypes.Bytes32
+
+	syncInterval time.Duration
+
+	mu        *sync.Mutex
+	syncCtxCf context.CancelFunc
+}
+
+func newPeerGroupDialer(
+	lggr logger.Logger,
+	peerGroupFactory rmn.PeerGroupFactory,
+	rmnHomeReader ccipreaderpkg.RMNHome,
+	bootstrapLocators []commontypes.BootstrapperLocator,
+	oraclePeerIDs []ragep2ptypes.PeerID,
+	commitConfigDigest [32]byte,
+) *peerGroupDialer {
+	return &peerGroupDialer{
+		lggr: lggr,
+
+		peerGroupFactory: peerGroupFactory,
+		rmnHomeReader:    rmnHomeReader,
+
+		bootstrapLocators:  bootstrapLocators,
+		oraclePeerIDs:      oraclePeerIDs,
+		commitConfigDigest: commitConfigDigest,
+
+		activePeerGroups: []rmn.PeerGroup{},
+
+		syncInterval: time.Minute, // todo: make it configurable
+
+		mu:        &sync.Mutex{},
+		syncCtxCf: nil,
+	}
+}
+
+func (d *peerGroupDialer) Start() {
+	if d.syncCtxCf != nil {
+		d.lggr.Warnw("peer group dialer already started, should not be called twice")
+		return
+	}
+
+	d.lggr.Infow("Starting peer group dialer")
+
+	ctx, cf := context.WithCancel(context.Background())
+	d.syncCtxCf = cf
+
+	go func() {
+		d.sync()
+
+		syncTicker := time.NewTicker(d.syncInterval)
+		for {
+			select {
+			case <-syncTicker.C:
+				d.sync()
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+}
+
+func (d *peerGroupDialer) Close() error {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	d.closeExistingPeerGroups()
+
+	if d.syncCtxCf != nil {
+		d.syncCtxCf()
+	}
+
+	return nil
+}
+
+func (d *peerGroupDialer) sync() {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	if !d.shouldSync() {
+		d.lggr.Debugw("No need to sync peer groups")
+		return
+	}
+	d.lggr.Infow("Syncing peer groups")
+
+	d.closeExistingPeerGroups()
+
+	if err := d.createNewPeerGroups(); err != nil {
+		d.lggr.Errorw("failed to create new peer groups", "err", err)
+		d.closeExistingPeerGroups() // close potentially opened peer groups
+	}
+}
+
+func (d *peerGroupDialer) shouldSync() bool {
```

Comment:
This needs unit tests, maybe make it a free func and pass in active and candidate digests read from the reader.